tracr/craft/bases.py

```diff
@@ -69,15 +81,31 @@ def __post_init__(self):
           f"of basis directions. Was {len(self.basis_directions)} "
           f"and {self.magnitudes.shape[-1]}.")
 
-    sort_idx = np.argsort(self.basis_directions)
-    self.basis_directions = [self.basis_directions[i] for i in sort_idx]
-    self.magnitudes = np.take(self.magnitudes, sort_idx, -1)
+    if not self.basis_is_sorted:
+      sort_idx = np.argsort(self.basis_directions)
+      self.basis_directions = [self.basis_directions[i] for i in sort_idx]
+      self.magnitudes = np.take(self.magnitudes, sort_idx, -1)
+      self.basis_is_sorted = True
+
+  def get_basis_set(self):
```

Comment:
Here and elsewhere: I think these getters should actually use `functools.cached_property`

Reply:
That's a very neat solution, I'll refactor everything to use this and omit the get methods and constructor parameters